util/Mongoose.hx

```diff
@@ -312,21 +325,22 @@ class Mongoose {
 		return objDecl;
 	}
 
-	static function classFieldToSchemaField( f : ClassField ){
+	static function classFieldToSchemaField( f : ClassField, typeKey:String ){
 
 		var type = {
 			pos : f.pos,
-			expr : typeToSchemaType(f.type)
+			expr : typeToSchemaType(f.type, typeKey)
 		};
-
-		var expr = macro { type : $type };
+        
+        if(typeKey.substr(0, 1) == "$")
+            typeKey = "@$__hx__" + typeKey; 
```

Comment:
Ok I see.
How about assuming field names are valid, and let the compiler fail otherwise ?